packages/react/src/auto/shared/AutoRichTextInput.tsx

```diff
@@ -1,49 +1,72 @@
-import {
-  BlockTypeSelect,
-  BoldItalicUnderlineToggles,
-  CodeToggle,
-  CreateLink,
-  DiffSourceToggleWrapper,
-  ListsToggle,
-  MDXEditor,
-  Separator,
-  UndoRedo,
-  diffSourcePlugin,
-  headingsPlugin,
-  linkDialogPlugin,
-  listsPlugin,
-  markdownShortcutPlugin,
-  quotePlugin,
-  thematicBreakPlugin,
-  toolbarPlugin,
-  type MDXEditorMethods,
-  type MDXEditorProps,
-} from "@mdxeditor/editor";
-import "@mdxeditor/editor/style.css";
 import type { ForwardedRef } from "react";
-import React, { useEffect, useRef } from "react";
-import { useFormContext, type Control } from "react-hook-form";
+import React, { useEffect, useRef, useState } from "react";
+import type { Control } from "react-hook-form";
+import { useFormContext } from "react-hook-form";
 import { get } from "../../utils.js";
 import { useStringInputController } from "../hooks/useStringInputController.js";
 import { multiref } from "../hooks/utils.js";
 
-export default function AutoRichTextInput(
-  props: {
-    field: string;
-    control?: Control<any>;
-    editorRef?: ForwardedRef<MDXEditorMethods> | null;
-  } & Omit<MDXEditorProps, "markdown" | "onChange">
-) {
+interface MDXEditorMethods {
+  setMarkdown: (markdown: string) => void;
+}
+
+interface AutoRichTextInputProps {
+  field: string;
+  control?: Control<any>;
+  editorRef?: ForwardedRef<MDXEditorMethods> | null;
+}
+
+const AutoRichTextInput: React.FC<AutoRichTextInputProps> = (props) => {
   const { formState } = useFormContext();
   const { field, control, editorRef, ...rest } = props;
   const controller = useStringInputController({ field, control });
   const innerRef = useRef<MDXEditorMethods>(null);
+  const [isEditorLoaded, setIsEditorLoaded] = useState(false);
+  const [mdxModule, setMdxModule] = useState<any>(null);
 
   useEffect(() => {
-    if (innerRef.current) {
+    const loadEditor = async () => {
+      try {
+        const module = await import("@mdxeditor/editor");
+        setMdxModule(module);
+        setIsEditorLoaded(true);
+      } catch (error) {
+        console.warn("Optional dependency not found, some features may not be available.");
+      }
+    };
+
+    void loadEditor();
+  }, []);
+
+  useEffect(() => {
+    if (innerRef.current && isEditorLoaded) {
       innerRef.current.setMarkdown(controller.value?.markdown ?? "");
     }
-  }, [controller.value]);
+  }, [controller.value, isEditorLoaded]);
+
+  if (!isEditorLoaded || !mdxModule) {
+    return <div>Loading editor...</div>;
```

Comment:
Did you want to try to use suspense to allow users to customize this loading state?